Integrations/MongoDB/MongoDB.py

```diff
@@ -0,0 +1,312 @@
+        return collection_object.delete_many(filter)
+
+    def create_collection(self, collection) -> Collection:
+        return self.db.create_collection(collection)
+
+    def drop_collection(self, collection):
+        return self.db.drop_collection(collection)
+
+
+def convert_id_to_object_id(entries: Union[List[dict], dict]) -> Union[List[dict], dict]:
+    """ Converts list or dict with `_id` key of type str to ObjectID
+
+    Args:
+        entries: The object contains list or dict with possible `_id` key.
+
+    Returns:
+        All '_id` key converted to ObjectID
+
+    Examples:
+        >>> convert_id_to_object_id([{'_id': '5e4412f230c5b8f63a7356ba'}])
+        [{'_id': ObjectId('5e4412f230c5b8f63a7356ba')}]
+
+        >>> convert_id_to_object_id({'_id': '5e4412f230c5b8f63a7356ba'})
+        {'_id': ObjectId('5e4412f230c5b8f63a7356ba')}
+    """
+    def _convert(entry: dict):
+        if "_id" in entry:
+            entry["_id"] = ObjectId(entry["_id"])
+        return entry
+
+    if isinstance(entries, list):
+        return [_convert(entry) for entry in entries]
+    return _convert(entries)
+
+
+def convert_object_id_to_str(entries: List[ObjectId]) -> List[str]:
+    """ Converts list of ObjectID to list of str.
+
+    Args:
+        entries: list of ObjectIDs
+
+    Returns:
+        List of strs representing the ObjectID
+
+    Examples:
+        >>> convert_object_id_to_str([ObjectId('5e4412f230c5b8f63a7356ba')])
+        ['5e4412f230c5b8f63a7356ba']
+    """
+    return [str(object_id) for object_id in entries]
+
+
+def test_module(client: Client, **kwargs) -> Tuple[str, dict]:
+    try:
+        _ = client.db.list_collection_names()
+    except OperationFailure as e:
+        return_error(str(e))
+    # To the unpack in the return.
+    return "ok", {}
+
+
+def get_entry_by_id_command(
+    client: Client, collection: str, object_id: str, **kwargs
+) -> Tuple[str, dict, list]:
+    raw_response = client.get_entry_by_id(collection, argToList(object_id))
+    readable_outputs = tableToMarkdown(
+        f"Total of {len(raw_response)} found in MongoDB collection `{collection}`:",
+        raw_response,
+    )
+    outputs = {CONTEXT_KEY: raw_response}
+    return readable_outputs, outputs, raw_response
+
+
+def search_query(
+    client: Client, collection: str, query: str, **kwargs
+) -> Tuple[str, dict, list]:
+    # test if query is a valid json
+    try:
+        query_json = json.loads(query)
+        raw_response = client.query(collection, query_json)
+        readable_outputs = tableToMarkdown(
+            f"Total of {len(raw_response)} found in MongoDB collection `{collection}` with query: {query}:",
+            [entry.get("_id") for entry in raw_response],
+            headers=["_id"],
+        )
+        outputs = {CONTEXT_KEY: raw_response}
+        return readable_outputs, outputs, raw_response
+    except JSONDecodeError:
+        return_error("MongoDB: the `query` parameter is not a valid json.")
```

Comment:
raise exception instead

Reply:
but the return_error is already doing it. Why should I raise an exception instead?